### DISCUSSION TITLE
How to write a Issue on Github( Bug Report )?

### DISCUSSION THREAD

**Discussion by kimjoyc** [janash/MolecularNodes#4] [General]
**Describe the bug**
Blender crashes when some molecular dynamics trajectories are loaded through MolecularNodes with an import filter. 

**To Reproduce**
Load the DNA system with the default import filter and attach the prmtop and dcd files from the molecular-nodes-bug repository link below located in the **Additional context** section.  Please refer to the repository to reproduce the bug for all test cases. 

The test case below involves a bond conflict with DNA where selection of resid 20 that leads to crashing, due to partial molecule selection unaccounted for in the atoms indices versus the bond indices. 

**Case 5 - Blender Crashes**
Make the following choices when importing:
- [ ] Choose "Ball and Stick"
- [ ] Type resid 20 in the default selection box.
> resid 20 represents the last DNA base pair. It will be bonded to resid 19

**Expected behavior**
Expect Blender to not crash and to load the trajectory.

**Error Codes**
Please include associated error codes that are shown in the python console when the error occurs.
> To view the python console on Windows, `Window` -> `Toggle System Console`, on Mac you must start Blender from the terminal.

Blender is crashing with exit code 139 and is found by going into Blender environment running ./blender executable on the terminal then following one of the test cases to make it crash. Then error is retraced with echo $? that yields 139. The Exit Code 139 means that the there is a SIGSEGV from Blender. It is most likely due to the bond conflicts relative to the all the atoms present.

```
# paste error codes from python console here
```

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Windows / MacOS / Linux]
 - Hardware: [e.g. Intel Mac, M1 Mac, x86 PC]
 - Blender Version: 3.4
- MolecularNodes Version: 2.4.1

**Additional context**

molecular-nodes-bug repository link below:
https://github.com/janash/molecular-nodes-bug 

 **Note: There is an incoming PR fix for the bug.**

**Comment by janash** [comment]
Hey @kimjoyc - I don't quite understand the comment. You should take the template and erase the directions and then write in your answers to the questions in the directions. For example, the bug report template says:

```
**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.
```

Yours would look something like this:

```
**Describe the bug**
Blender crashes when some molecular dynamics trajectories are loaded through MolecularNodes with an import filter. (you can write something else here)

**To Reproduce**
Here you would describe loading the DNA system with the default import filter. Attach the prmtop and dcd files to the issue and refer to them here to allow the maintainer to reproduce the bug. You can describe *one* test case here (the simplest one). If you want to put all five test cases, add them below the template.

**Expected behavior**
Expect Blender to not crash and to load the trajectory.
```

For the Error Codes section, describe that you don't see a Python error code. Instead, you can tell that Blender is crashing with exit code 139. You can describe how you found this out through `echo $?`

I believe it is possible to attach files to an issue. In this case, attach the test systems.  If not, make a GitHub repo containing the files and link to it.

**Comment by kimjoyc** [comment]
Ok thanks. I thought I needed to cover everything from your bug repo in the issue. I'll rework this and link your repo for extra details.

**Comment by janash** [comment]
You can explain all of the test cases, but I'd advise putting them at the end of the report after all of the template information. Put just one example in the reproducing section and then say there are more test cases given below. You can also indicate that a PR with a fix is coming.

**Reply by kimjoyc** [reply]
Fixed , let me know if there is something else to add.

**Comment by janash** [comment]
You can remove this sentence " For instance, the default selection deselects certain water." since we didn't provide evidence. You can also add a comment at the end saying a PR with a solution is coming. Otherwise, the issue LGTM and you can file it :) Thanks!
